File: 35-search-insert-position/35-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int low = 0;
        int high = nums.size() - 1;
        int middle;
        
        while(low <= high){
            middle = low + (high - low)/2;
            if(nums[middle] == target)
                return middle;
            else if(nums[middle] < target)
                low = middle  + 1;
            else
                high = middle - 1;
        }
        return low;
    }
};

File: 35-search-insert-position/35-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        low: int = 0
        high: int = len(nums) - 1
        
        while low <= high:
            middle: int = low + int((high - low)/2)
            if nums[middle] == target:
                return middle
            elif nums[middle] < target:
                low = middle  + 1
            else:
                high = middle - 1
            
        return low